app/helpers/application_helper.rb

```diff
@@ -1,3 +1,28 @@
 # frozen_string_literal: true
 module ApplicationHelper
+  def admin?
+    return if current_user && current_user.admin?
+    redirect_to_current_or_root
+    flash[:alert] = 'Access denied'
+  end
+
+  def moderator?
+    return if current_user && current_user.moderator?
+    redirect_to_current_or_root
+    flash[:alert] = 'Access denied'
+  end
+
+  def admin_or_moderator?
+    return if current_user && (current_user.admin? || current_user.moderator?)
+    redirect_to_current_or_root
+    flash[:alert] = 'Access denied'
+  end
+
+  def redirect_to_current_or_root
```

Comment:
you can remove this method and use http://blog.bigbinary.com/2016/02/29/rails-5-improves-redirect_to_back-with-redirect-back.html